pkg/controller/operands/operand.go

```diff
@@ -93,7 +93,14 @@ func (h *genericOperand) ensure(req *common.HcoRequest) *EnsureResult {
 	found := h.hooks.getEmptyCr()
 	err = h.Client.Get(req.Ctx, key, found)
 	if err != nil {
-		return h.createNewCr(req, err, cr, res)
+		e := h.createNewCr(req, err, cr, res)
+		if e.Err != nil && strings.Contains(e.Err.Error(), "already exists") {
+			res.Err = nil
+			found.SetName(cr.GetName())
```

Comment:
I fear that if the cache is not correctly populated for our object due to missing selectors, also the get on line 94 would fail and so we don't have found at all.
Probably the best and simplest strategy is:
- try creating as now,
- then, if it fails with IsAlreadyExists(err)=true, build a new client using New ( https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go#L75 ) from controller runtime client to get a client that completely bypasses the cache on reads and writes
- and finally use that client to fetch the object again and update it

Reply:
cluster.APIReader should also work:
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/cluster/cluster.go#L69-L72